src/Microsoft.AspNetCore.Sockets.Http/Internal/Transports/WebSocketsTransport.cs

```diff
@@ -138,8 +139,21 @@ public WebSocketsTransport(WebSocketOptions options, IDuplexPipe application, De
         {
             try
             {
+                var endOfMessage = true;
                 while (true)
                 {
+#if NETCOREAPP2_1
+                    // If there was a read that had a 'false' EndOfMessage then we should skip the 0 byte read since there is an in progress frame
+                    if (endOfMessage)
```

Comment:
Not sure how to cleanly not copy this code from below

Reply:
If we can unify more of the connection primitives, we may be able to unify the transports a bit. Let's not worry too much about this right now, as long as we have testing in place for behavior changes and automated benchmarks for performance changes like this to catch regression.